Elements/src/Validators/Validators.cs

```diff
@@ -348,15 +348,21 @@ public class PolylineValidator : IValidator
 
         public void PostConstruct(object obj)
         {
+            if(obj is Polygon) {
+                // we don't need to validate twice — 
+                // the Polygon PostConstruct validator will handle all of this correctly.
+                return;
+            }
+            var polyline = obj as Polyline;
+            polyline.Vertices = Vector3.RemoveSequentialDuplicates(polyline.Vertices);
```

Comment:
I always worried that we'd put too much logic into these `PostConstruct`/`PreConstruct` methods and forget that these constructors are _also_ used for JSON deserialization. Is it logical to have a deserialization process that might mutate the incoming polyline? Assuming that Elements is doing most of the creation of these Polylines (perhaps a big assumption), do we really want to do this sequential vertex removal _again_ when we deserialize?

Reply:
I said all of the above, without suggesting an alternative, which you then described and its sounds worse from a usability perspective to put this into its own constructor which a developer would have to know how to find. Let's leave your change as is.